Enterprise-CMCS/managed-care-review
# MCR-3609: State indicate not rates in rate cert

## Summary
This PR 

- Replaces the "Programs this rate certification covers" field with "What rates are included in this certification?" on Rate Details, ReviewSubmit and SubmissionSummary
- Updates the rateRevision table so that previous submissions rateProgramIDs are renamed as the `deprecatedRateProgramIDs` field and creates a new column called `rateProgramIDs` this is the same name as previous 
- Updates the statePrograms.json to include new programs that are rate programs
- updates the Toph user to be from FL since that is a state that now has rate programs
- Adds a new field on programSelect to determine if all programs or only contract programs should be displayed

#### Related issues
https://jiraent.cms.gov/browse/MCR-3609
#### Screenshots

Submission Summary Page 
<img width="718" alt="Screenshot 2024-05-15 at 4 45 26 PM" src="https://github.com/Enterprise-CMCS/managed-care-review/assets/67110378/509877df-4e80-42ed-b710-6471b3ac08f3">

Field Validation on Rate Details
<img width="454" alt="Screenshot 2024-05-14 at 11 11 29 AM" src="https://github.com/Enterprise-CMCS/managed-care-review/assets/67110378/4385f354-fb92-4715-9765-79bec19d70d3">
<img width="406" alt="Screenshot 2024-05-14 at 11 11 35 AM" src="https://github.com/Enterprise-CMCS/managed-care-review/assets/67110378/09fccc38-98d2-4ad0-9db4-a36123734254">


Rate Details page includes both rate and contract programs
<img width="381" alt="Screenshot 2024-05-17 at 2 23 37 PM" src="https://github.com/Enterprise-CMCS/managed-care-review/assets/67110378/d88c4e62-fa3d-4cd6-9cf8-630383f6f87c">

Contract page includes only contract programs

<img width="420" alt="Screenshot 2024-05-17 at 2 22 57 PM" src="https://github.com/Enterprise-CMCS/managed-care-review/assets/67110378/3775e326-cbb0-40fb-a83d-3e9ef49b6d5e">


#### Test cases covered

- Program select doesnt include rate programs when `contractProgramsOnly` is true
- Submission summary displays the deprecated field on submissions that have a populated `rate.deprecatedRateProgramIDs` field

## QA guidance

- Sign in as Toph
- Create a new submission. Note that the Rate Details page includes the contract programs and rate programs indicated in the [state program sheet](https://docs.google.com/spreadsheets/d/138HHJTs4sGZucyffNFDe2dlSvYxPRSrP/edit#gid=687010743)
- Rate Details displays the new wording for the field and displays an updated validation warning if left blank
- ReviewSubmit page displays "Rates this rate certification covers" instead of "programs this rate cert covers"
- For CMS users The SubmissioinSummary page displays "Rates this rate certification covers" instead of "programs this rate cert covers"
